### DISCUSSION TITLE
Recurring trim on detached volume

### DISCUSSION THREAD

**Discussion by rasmusson** [longhorn/longhorn#8035] [Q&A]
I have a volume that is only used 5 min per day by a cron job. The rest of the time it shows up as detached. 
Every night I do a filesystem trim of all volumes, but this one fails saying:

... level=fatal msg="Failed to do a recurring job" func=main.main.RecurringJobCmd.func4 file="recurring_job.go:76" error="failed to complete filesystem-trim for :  Action [trimFilesystem] not available on [&{pvc-0f7bb829-ff84-482d-af09-19c1ce8d4fde...

I guess this is because it is detached. How can I have the recurring job trimming my cron job volume?

**Comment by derekbit** [comment]
@rasmusson 
Yes, Longhorn doesn't support recurring job trimming volumes for now.
Ref: https://github.com/longhorn/longhorn-manager/blob/master/api/model.go#L1513-L1571


@shuo-wu @innobead Do you think we should support trimming detached volumes? If a volume is detached and no data is written into the volume, trimming it seems meaningless.

**Reply by rasmusson** [reply]
Ok, I see that its not supported right now but I don't understand you second comment. Plenty of data is written when I run the cronjob. Is there no need to trim a volume that is not attached all the time?

**Reply by derekbit** [reply]
OK, understand your application now. Then, I will open a feature request for the trim job.

**Reply by derekbit** [reply]
Let's track in https://github.com/longhorn/longhorn/issues/8052

**Reply by rasmusson** [reply]
Thank you!

**Reply by rasmusson** [reply]
Any suggestions on workarounds? On the milestone tag it sound like it could take a while.
